backend/validators/streaks.py

```
@@ -0,0 +1,142 @@
+    return bool(
+        snap is not None
+        and snap.status == 'active'
+        and snap.metrics_samples >= 1
+        and snap.logs_samples >= 1
+        and snap.metrics_status != 'shame'
+        and snap.logs_status != 'shame'
+        and snap.version_status != 'shame'
+    )
+
+
+def _has_observation(snap):
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_has_observation`

(ANN202)